packager/VisibilityChecker.cc

```diff
@@ -439,13 +439,29 @@ class VisibilityCheckerPass final {
         auto importType = this->package.importsPackage(otherPackage);
         if (!importType.has_value()) {
             // We failed to import the package that defines the symbol
+            bool layeringViolation = false;
+            auto &pkg = ctx.state.packageDB().getPackageInfo(otherPackage);
+            auto strictDepsLevel = this->package.strictDependenciesLevel();
+            std::string errorMessage;
+            bool isTestImport = otherFile.data(ctx).isPackagedTest() || ctx.file.data(ctx).isPackagedTest();
+            if (!isTestImport && db.enforceLayering() && strictDepsLevel.has_value() &&
+                strictDepsLevel.value().first != core::packages::StrictDependenciesLevel::False &&
+                this->package.causesLayeringViolation(db, pkg)) {
+                layeringViolation = true;
+                errorMessage = core::ErrorColors::format("`{}` resolves but its package cannot be imported because "
+                                                         "importing it would cause a layering violation",
+                                                         lit.symbol.show(ctx));
+            } else {
+                errorMessage =
+                    core::ErrorColors::format("`{}` resolves but its package is not imported", lit.symbol.show(ctx));
+            }
```

Comment:
Can we move this all under the `if (auto e = ...)` so that we only evaluate it if we're reporting errors for `MissingImport`?  This is a small nit, but it's more in line with how other errors are handled -- and it enables us to not use `ErrorColors::format`, but call `setHeader` in the conditionals directly, etc.